docs/dev/reports/README.md

```diff
@@ -0,0 +1,469 @@
+# Reports system design docs
+
+This document summarizes the reports system design.
+
+## Intro
+
+Kyverno generates reports based on policies and matching resources.
+Every policy is evaluated against matching resources and the result of the evaluation is recorded in reports.
+
+This happens in two ways:
+- at admission time, the policies matching the resource being admitted are evaluated
+- in the background, policies are periodically evaluated against matching resources
+
+Admission time reports and background scan reports have a very similar structure but a very different lifecycle.
+
+Reports that are created at admission time and in the background are intermediary reports used internally by Kyverno to produce higher level reports:
+- policy reports/cluster policy reports, aggregated per namespace for namespaced resources
+- cluster policy reports, aggregated at the cluster level for cluster wide resources
+
+Keep in mind that only validation and image verification rules are recorded in the reports, mutation and generation rules are not.
```

Comment:
I was under the impression, that we record pass / skip / fail for mutate and generate as well. Seems like that would be required for compliance.

Reply:
We have never done this. I opened #3837 a year ago to do exactly that.